docs/open_source/testset_generation/rag_evaluation/index.md

```diff
@@ -141,6 +141,239 @@ Built-in metrics include `ragas_context_precision`, `ragas_faithfulness`, `ragas
 
 Alternatively, you can directly pass a list of answers instead of `get_answer_fn` to the `evaluate` function, you can then pass the retrieved documents as an optional argument `retrieved_documents` to compute the RAGAS metrics.
 
+### Custom Metrics
+
+**You can also implement your own metrics** using the base `Metric` class from Giskard.
+
+Here is an example of how you can implement a custom LLM-as-judge metric, as described in the [RAG cookbook](https://huggingface.co/learn/cookbook/en/rag_evaluation#evaluating-rag-performance) by Huggingface.
+
+#### Step 1 - Subclass the Metric class
+
+Implement the new metric in the `__call__` method: 
```

Comment:
Perhaps we can guide the user a bit better through the process and things to consider. For example, the name of the metric and key in the dict should be the same.